### PR TITLE
CASEFLOW-1749: Support adding substitute appellants for post-dispatch cases

### PR DESCRIPTION
**Pull request by n1zyy**
### Description
Resolves [CASEFLOW-1749](https://vajira.max.gov/browse/CASEFLOW-1749).

Previously, the "Add Substitute" button was only shown if an appeal was in "dispatched" state. If an open task was added after that, the task went to "post_dispatch" state, and the button was no longer shown.

Now, both states are allowed, so that adding a substitute appellant is not barred if tasks are added to a dispatched appeal.

### Acceptance Criteria
(lifted from Jira)

- [ ] As a CAVC Litigation Support user, I need the ability to establish a CAVC appeal even if there are open tasks on the previously dispatched appeal, so that appeal can enter the Boards inventory/workflow and be decided.  
- [ ] the user is able to see the add CAVC button on a dispatched appeal with active tasks
- [ ] the original appeals tasks does not block distribution of the CAVC appeal
- [ ] the original appeals tasks does not block the dispatch of the CAVC appeal

### Testing Plan
1. You can make use of the new trait I added to easily create a post-dispatch appeal with a decision issue. You should now see the "+ Add Substitute" button when logged in as COB_USER.

#### Helpful tips:

You can create a post-dispatch appeal like so:

```ruby
a = FactoryBot.create(:appeal, :post_dispatch, :with_decision_issue,
   docket_type: "evidence_submission",
   stream_type: "original",
   disposition: "dismissed_death")
```

If you're doing more than testing for the presence of the button, you'll want to add a date of death to the veteran:

```ruby
v = a.veteran
v.date_of_death = Date.today - 1.year
v.save
```

**N.B. - It may be necessary to run `make clean` to clear the webpack cache before testing locally or running tests.**

### CODE REVIEW

File: client/app/queue/CaseDetailsView.jsx [hunk 1]
```diff
@@ -144,10 +144,10 @@ export const CaseDetailsView = (props) => {
     [appeal, tasks]
   );
 
-  const appealIsDispatched = appeal.status === 'dispatched';
+  const appealIsDispatched = ['dispatched', 'post_dispatch'].includes(appeal.status);
 
   const editInformation =
-  appeal.appellantType !== 'VeteranClaimant' && props.featureToggles.edit_unrecognized_appellant;
+    appeal.appellantType !== 'VeteranClaimant' && props.featureToggles.edit_unrecognized_appellant;
```
Comment (n1zyy) [inline client/app/queue/CaseDetailsView.jsx:150]: Unrelated, but this annoyed me. 👼

File: db/seeds/substitutions.rb [hunk 2]
```diff
@@ -30,7 +30,7 @@ def create_appeal_with_death_dismissal(veteran: deceased_vet, docket_type: "dire
 
       create(
         :appeal,
-        :dispatched_with_decision_issue,
+        :dispatched, :with_decision_issue,
```
Comment (n1zyy) [inline db/seeds/substitutions.rb:33]: "dispatched_with_decision_issue" simply extended "dispatched" and added a decision issue, so this probably should have been two separate things from the start.

I needed to do "post-dispatch with decision issue" so I split "with decision issue" into its own trait -- and then had to fix up existing uses of "dispatched with decision issue" like this one.

File: spec/factories/appeal.rb [hunk 3]
```diff
@@ -368,6 +368,17 @@
       end
     end
 
+    # An appeal which was dispatched, but has then had other open tasks added.
+    # Note that the -ed suffix in 'dispatched' does not carry over to 'post_dispatch', which is how
+    # it is referred to elsewhere in the code.
```
Comment (n1zyy) [inline spec/factories/appeal.rb:373]: Is this ambiguous? I mean to say that the naming used here ("dispatched" vs "post_dispatch") is consistent. Do I even need to clarify this? Am I too indecisive?

File: client/app/queue/CaseDetailsView.jsx [hunk 4]
```diff
@@ -144,10 +144,10 @@ export const CaseDetailsView = (props) => {
     [appeal, tasks]
   );
 
-  const appealIsDispatched = appeal.status === 'dispatched';
+  const appealIsDispatched = ['dispatched', 'post_dispatch'].includes(appeal.status);
```
Comment (n1zyy) [inline client/app/queue/CaseDetailsView.jsx:147]: The actual fix, easy-peasy.

File: spec/feature/queue/case_details_spec.rb [hunk 5]
```diff
@@ -1935,6 +1936,13 @@
           it_behaves_like "the button is not shown"
         end
 
+        context "When the appeal is in post-dispatch state" do
+          let(:disposition) { "dismissed_death" }
+          let(:status) { :post_dispatch }
+
+          it_behaves_like "the button is shown"
+        end
+
```
Comment (n1zyy) [inline spec/feature/queue/case_details_spec.rb:1945]: The actual test for the fix, easy-peasy.

**Review by jcq — APPROVED** [review]
This all seems straightforward. Tested manually and LGTM!

**Pull request merged**